src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,118 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        this.table = (Node<K,V>[]) new Node[DEFAULT_CAPACITY];
+        this.threshold = (int)(LOAD_FACTOR * DEFAULT_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        int bucketIndex = getBucketIndex(hash);
+        Node<K, V> bucket = table[bucketIndex];
+        if (bucket == null) {
+            table[bucketIndex] = new Node<>(hash, key, value, null);
+        } else {
+            while (bucket.next != null) {
+                if (bucket.hash == hash && Objects.equals(bucket.key, key)) {
+                    bucket.value = value;
+                    return;
+                }
+                bucket = bucket.next;
+            }
+            if (bucket.hash == hash && Objects.equals(bucket.key, key)) {
+                bucket.value = value;
+                return;
+            }
+            bucket.next = new Node<>(hash, key, value, null);
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> bucket = table[getBucketIndex(hash(key))];
+        while (bucket != null) {
+            if (Objects.equals(bucket.key, key)) {
+                return bucket.value;
+            }
+            bucket = bucket.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        int newCapacity = table.length << 1;
+        Node<K, V>[] oldTable = table;
+        table = (Node<K,V>[]) new Node[newCapacity];
+        threshold = (int)(LOAD_FACTOR * newCapacity);
+        transferAllNodesToTable(oldTable);
```

Comment:
this method can be simplified by calling put for each bucket that is not zero in the old table

Reply:
we should definitely reuse our `put()` method here, the same thing as with the `ArrayList`, but this time we also need to iterate over the nodes inside our bucket, and through the buckets as well.
So it should be only 2 loops inside one another, and they can be inside this `resize()` method